mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -197,6 +200,21 @@ LogicalResult Conv2dGenerator::isValidConvolution() const {
   return success();
 }
 
+LogicalResult Conv2dGenerator::isValidChip() const {
+  // We support xdlops iff it is a gfx908 chip, fail otherwise
+  if (config.xdlops && config.chip != "gfx908")
+    return failure();
+
+  // We support up to gfx908 for nonxdlops algorithm
+  // For example, both gfx90a and gfx1030 are unsupported now
+  unsigned int chipHexNumber = 0;
+  sscanf(config.chip.c_str(), "gfx%x", &chipHexNumber);
+  if (chipHexNumber > 0x908)
+    return failure();
```

Comment:
Since we're here, might as well reject chips older than 0x900, which'll also catch a sscanf() failure?